osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/infra/messaging/DlmsConnectionMessageProcessor.java

```diff
@@ -58,13 +57,18 @@
 
     public DlmsConnectionManager createConnectionForDevice(final DlmsDevice device,
             final MessageMetadata messageMetadata) throws OsgpException {
-        
+
         throttlingService.openConnection();
 
         final DlmsMessageListener dlmsMessageListener = this
                 .createMessageListenerForDeviceConnection(device, messageMetadata);
 
-        return this.dlmsConnectionHelper.createConnectionForDevice(device, dlmsMessageListener);
+        try {
```

Comment:
this is the fix, if the connection fails somehow, release the slot. There is a postProcessing method also that releases the connection, but it checks it the connection was null and then returns, but of course in case of connection failure the connection is null, so the slot was never released.